test/routes/issueRoutesTest.js

```diff
@@ -0,0 +1,317 @@
+                done();
+            });
+            
+            it('should edit the issue', function(done) {
+                request('http://localhost:8000')
+                .put('/issues/5aba5d0301f3703b5c720459')
+                .send(issue)
+                .then((response) => {
+                    
+                    const createdIssue = response.body;
+
+                    expect(createdIssue).to.be.an('object');
+                    expect(createdIssue.title).to.equal(issue.title);
+                    expect(createdIssue.description).to.equal(issue.description);
+                    expect(createdIssue.name).to.equal(issue.name);
+
+                    done();
+                }, done)
+                .catch(done);
+            });
+        });
+        describe('Invaild params', function() {
+
+            let issue;
+            before((done) => {
+                issue = {
+                    title: 0,
+                    name: null
+                };
+                done();
+            });
+    
+            it('should not edit issue', function(done){
+                request('http://localhost:8000')
+                .put('/issues/5aba5d0301f3703b5c72045')
+                .send(issue)
+                .then(done, (err) => {
+                    expect(err).to.not.be.null;
+                    expect(err).to.have.status(500);
+                    done();
+                })
+                .catch(done);
+            });
+        });
+    });
+    describe('Delete issue', function(){        
+
+        describe('Valid params', function() {
+
+            let issue;
+
+            it('should delete the issue', function(done) {
+                request('http://localhost:8000')
+                .delete('/issues/5aba5d0301f3703b5c720459')
+                .then((response) => { 
+                    issue = response.body;
+                    expect(issue).to.be.an('object');                    
+                })
+                .then((issue) => {
+                    expect(issue).to.be.undefined;
+                    done();
+                })
+                .catch(done);
+            });
+        });
+        describe('Invaild params', function() {
+
+            it('should not delete issue', function(done){
+                request('http://localhost:8000')
+                .delete('/issues/5aba5d0301f3703b5c72045')
+                .then(done, (err) => {
+                    expect(err).to.not.be.null;
+                    expect(err).to.have.status(500);                    
+                    done();
+                })
+                .catch(done);
+            });
+        });
+    });
+    describe('Mark complete issue', function(){        
+
+        describe('Valid params', function() {
+            
+            it('should mark complete the issue', function(done) {
+                request('http://localhost:8000')
+                .patch('/issues/5ab25c27f608fb1f201413e5/markCompleted')
```

Comment:
Sometimes having an explicit url `issue/<issue_id>/markComplete` is fine. In this case we want to have only 

PATCH /issue/<issueID>/ and then let the params decide which attributes to update. This will also eliminate the need to have couple different endpoints like 'markComplete', 'markPending', etc.